libraries/AP_TECS/AP_TECS.cpp

```diff
@@ -1168,22 +1163,28 @@ void AP_TECS::_initialise_states(int32_t ptchMinCO_cd, float hgt_afe)
         _hgt_dem              = hgt_afe;
         _hgt_dem_in_prev      = hgt_afe;
         _hgt_dem_in_raw       = hgt_afe;
-        _TAS_dem_adj          = _TAS_dem;
-        _flags.reset          = true;
-        _post_TO_hgt_offset   = _climb_rate * _hgt_dem_tconst;
         _flags.underspeed     = false;
         _flags.badDescent     = false;
 
-        _max_climb_scaler = 1.0f;
-        _max_sink_scaler = 1.0f;
+        if (_in_takeoff_first_stage()) {
+            _post_TO_hgt_offset   = _climb_rate_limit * _hgt_dem_tconst; // Replacement prevents oscillating hgt_rate_dem.
+            _TAS_dem_adj = _TAS_dem;
+            _max_climb_scaler = 1.0f;
+            _max_sink_scaler = 1.0f;
+            _pitch_demand_lpf.reset(_ahrs.get_pitch());
+            _pitch_measured_lpf.reset(_ahrs.get_pitch());
+        }
 
-        _pitch_demand_lpf.reset(_ahrs.get_pitch());
```

Comment:
if we never do a takeoff (eg. done via FBWA) then this would mean no lpf resets

Reply:
@tridge this is already taken care of in L1174: https://github.com/ArduPilot/ardupilot/pull/27174/files#diff-30aee80148363a53a210bc83701e744855f97514d3ce986ee83b1c0dc8fdf569R1174

Reply:
There are dedicated resets for that case in the previous if-case, L1154.